File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.py
class Solution(object):
    def countPartitions(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total_sum = sum(nums)
        left_sum = 0 
        count = 0 
        for num in nums[:-1] :
            left_sum += num 
            total_sum -= num 
            if left_sum % 2 == 0 and total_sum % 2 == 0:
                count += 1 
            elif left_sum % 2 == 1 and total_sum % 2 == 1:
                count += 1 
        return count

File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.java
class Solution {
    public int countPartitions(int[] nums) {
        int res = 0 , totalSum = 0 , prefixSum = 0 ; 

        for (int num : nums) 
        {
            totalSum += num ; 
        }

        for (int i = 0  ; i <  nums.length - 1 ; i++ ) 
        {
            prefixSum += nums[i] ; 
            totalSum -= nums[i] ; 

            if ( (prefixSum - totalSum) % 2 == 0 ) res++ ; 
        }

        return res ; 
    }
}